woocommerce/woocommerce
# [Product Block Editor]: re implement product description field

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR re-implements the `Description` block. Most relevant changes:

* Now Description blocks support inner blocks
* Render a Summary block instance when the Description block hasn't any other block
* Extend the Description block toolbar with a new `Full editor` button
* It switches to the blocks preview component once the user sets the product description

~Disclaimer: I left the current `Add/Update description` button just for testing purposes. This PR requires https://github.com/woocommerce/woocommerce/pull/41859 for its full functionality.~

~It requires https://github.com/woocommerce/woocommerce/pull/41859~

Known / follow-up tasks

* ~Connect the UI with the store (https://github.com/woocommerce/woocommerce/pull/41859)~
* Propagate the initial content from the description block to the modal editor 
* Move the Modal Editor component out of the description block. Ideally, in the block editor file
* Tweak the layout/styles of the blocks preview


Part of https://github.com/woocommerce/woocommerce/issues/39622

Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to the product editor
2. Identify the Description section
3. Select the Description block
4. Confirm you see the `Full editor` button. ~This button doesn't work yet. It requires https://github.com/woocommerce/woocommerce/pull/41859~

<img width="707" alt="Screenshot 2023-12-04 at 16 31 30" src="https://github.com/woocommerce/woocommerce/assets/77539/8bc1b556-cf7b-4857-a7b0-55efaf9e6147">

6. Open the modal editor by clicking on the `Full editor` button
7. Set a content for the product description field
8. Click on `Done` button
9. Confirm it shows the regular preview component
10. Unselect and select the block again
11. Confirm now it shows a `Edit in full editor` toolbar button.

<img width="698" alt="Screenshot 2023-12-04 at 16 33 40" src="https://github.com/woocommerce/woocommerce/assets/77539/2abdbada-4811-4478-9b6b-a376007d300d">

12. Confirm clicking on the `Edit in full editor` opens the modal again



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
